Modules/_asynciomodule.c

```diff
@@ -193,6 +193,49 @@ is_coroutine(PyObject *coro)
     return has_it;
 }
 
+/*[clinic input]
+_asyncio.isfuture
+
+    obj: object
+
+Return True if obj is a Future instance.
+
+This returns True when obj is a Future instance or is advertising
+itself as duck-type compatible by setting _asyncio_future_blocking.
+See comment in Future for more details.
+
+[clinic start generated code]*/
+
+static PyObject *
+_asyncio_isfuture_impl(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=7ceb4ecaca613a86 input=b417c209efff25ec]*/
+{
+    _Py_IDENTIFIER(__class__);
```

Comment:
Good idea.

Reply:
@asvetlov @1st1 isfuture check if _asyncio_future_blocking exists and is not None because mock object has _asyncio_future_blocking as None and we want to return isfuture() as False in this case.
See test case https://github.com/python/cpython/blob/8425de4147eb8d83befbb8ea77516fc764bb4309/Lib/test/test_asyncio/test_futures.py#L123-L124

Do we really want to use Future_CheckExact to return fast?